backend/src/api/model/event.rs

```diff
@@ -519,6 +519,50 @@ impl AuthorizedEvent {
         Ok(event)
     }
 
+    pub(crate) async fn create_placeholder(event: NewEvent, context: &Context) -> ApiResult<Self> {
+        let query = format!("\
+            insert into all_events ( \
+                opencast_id, title, description, \
+                creators, series, tracks, \
+                read_roles, write_roles, preview_roles, \
+                metadata, is_live, updated, created, state \
+            ) values ( \
+                $1, $2, $3, $4, $5, $6, $7, $8, \
+                '{{}}', '{{}}', false, '-infinity', now(), 'waiting' \
+            ) returning id \
+        ");
+
+        let acl = convert_acl_input(event.acl);
+        let dummy_tracks = vec![EventTrack {
+            uri: "https://example.org/video.mp4".to_string(),
+            flavor: "presenter/preview".to_string(),
+            mimetype: Some("video/mp4".to_string()),
+            resolution: Some([1280, 720]),
+            is_master: Some(true),
+        }];
+
+        context.db.execute(&query, &[
+            &event.opencast_id,
+            &event.title,
+            &event.description,
+            &event.creators,
+            &event.series_id.map(|id| id.key_for(Id::SERIES_KIND)),
+            &dummy_tracks,
+            &acl.read_roles,
+            &acl.write_roles,
+        ]).await?;
+
+        let event = Self::load_by_opencast_id(event.opencast_id, context)
+            .await?
+            .ok_or_else(|| err::invalid_input!(
+                // TODO: change to generic key once #1313 is merged
+                key = "event.placeholder.not-found",
+                "event not found",
```

Comment:
This is merged now. But apart from that, I think you can just replace the `ok_or_else(...)` with `.unwrap()`. This is guaranteed to work, or something is seriously wrong with the database. It's certainly not an `invalid_input`.